src/services/offer.js

```diff
@@ -239,6 +239,16 @@ class OfferService {
         return offer;
     }
 
+    async getOfferByCompanyID(companyID, user, hasAdminPrivileges) {
+        let offers = await Offer.find({ owner: companyID });
+
+        offers = offers.filter((offer) =>
+            !(offer.isHidden && !(hasAdminPrivileges || offer.owner.toString() === user?.company?._id.toString()))
```

Comment:
Maybe we should have a private function with this verification, not that it's being used in two places https://github.com/NIAEFEUP/nijobs-be/blob/4335350ff2e146a7f26e6a14837863cde2fbb9e1/src/services/offer.js#L237

Reply:
I was wondering if we could reuse the `hasOwnershipRights` validator, but I don't think it is possible since it relies on the express (req,res,next) structure.

https://github.com/NIAEFEUP/nijobs-be/blob/develop/src/api/middleware/auth.js#L41

Anyway, I agree we should extract the common logic and use in the offer service and here. Otherwise, we could add a function in the offer service that would do the hasOwnershipRights verification (returns true or false), and use the .filter here. And in the offer service, it would call it's own function there as well.